sky/utils/controller_utils.py

```diff
@@ -800,8 +822,36 @@ def maybe_translate_local_file_mounts_and_sync_up(task: 'task_lib.Task',
         sources = list(src_to_file_id.keys())
         sources_str = '\n    '.join(sources)
         logger.info(f'  {colorama.Style.DIM}Files (listed below) '
-                    f' -> storage: {file_bucket_name}:'
+                    f' -> storage: {bucket_name}:'
                     f'\n    {sources_str}{colorama.Style.RESET_ALL}')
+
+    if bucket_wth_prefix is not None and len(new_storage_mounts) > 0:
+        # The full path from the user config of IBM COS contains the region,
+        # and Azure Blob Storage contains the storage account name, we need to
+        # check the region and storage account name is match with the system
+        # configured.
+        if store_type == storage_lib.StoreType.IBM.value:
+            store: storage_lib.IBMCosStore = list(  # type: ignore
+                new_storage_mounts.values())[0].stores[storage_lib.StoreType(
+                    store_type)]
+            _, _, region = data_utils.split_cos_path(bucket_wth_prefix)
+            assert store.region == region, (
+                f'The region from job config {bucket_wth_prefix} does '
+                f'not match the region of the storage sky supports '
+                f'{store.region}')
+        elif store_type == storage_lib.StoreType.AZURE.value:
+            store: storage_lib.AzureBlobStore = list(  # type: ignore
+                new_storage_mounts.values())[0].stores[storage_lib.StoreType(
+                    store_type)]
+            storage_account_name, _, _ = data_utils.split_az_path(
+                bucket_wth_prefix)
+            env_storage_account_name = \
+                store.storage_account_name  # type: ignore
+            assert storage_account_name == env_storage_account_name, (
```

Comment:
This is the issue: if we allow user support for the full Azure path, we need to either ignore it or raise it. 

Alternatively, we can expose an interface to set the account outside of the store object. @romilbhardwaj

Reply:
I set them in the init of storage function.